persim/images.py

```diff
@@ -210,3 +220,695 @@ def show(self, imgs, ax=None):
+        if out_file:
+            plt.savefig(out_file, bbox_inches='tight')
+
+def dict_print(dict_in):
+    # print dictionary contents in human-readable format
+    if dict_in is None:
+        str_out = 'None'
+    else:
+        str_out = []
+        for key, val in dict_in.items():
+            str_out.append('%s: %s' % (key, str(val)))
+        str_out = '{' + ', '.join(str_out) + '}'
+
+    return str_out
+
+def _transform(pers_dgm, skew=True, resolution=None, weight=None, weight_params=None, kernel=None, kernel_params=None, _bpnts=None, _ppnts=None):
+        """
+        Transform a persistence diagram to a persistence image using the parameters specified in the PersistenceImager
+        object instance
+        :param pers_dgm: (N,2) numpy array of persistence pairs encoding a persistence diagram
+        :param skew: boolean flag indicating if diagram needs to be converted to birth-persistence coordinates
+                     (default: True)
+        :return: numpy array encoding the persistence image
+        """
+        pers_dgm = np.copy(pers_dgm)
+        pers_img = np.zeros(resolution)
+        n = pers_dgm.shape[0]
+        general_flag = True
+
+        # if necessary convert from birth-death coordinates to birth-persistence coordinates
+        if skew:
+            pers_dgm[:, 1] = pers_dgm[:, 1] - pers_dgm[:, 0]
+
+        # compute weights at each persistence pair
+        wts = weight(pers_dgm[:, 0], pers_dgm[:, 1], **weight_params)
+
+        # handle the special case of a standard, isotropic Gaussian kernel
+        if kernel == bvncdf:
+            general_flag = False
+            sigma = kernel_params['sigma']
+
+            # sigma is specified by a single variance
+            if isinstance(sigma, (int, float)):
+                sigma = np.array([[sigma, 0.0], [0.0, sigma]], dtype=np.float64)
+
+            if (sigma[0, 0] == sigma[1, 1] and sigma[0, 1] == 0.0):
+                sigma = np.sqrt(sigma[0, 0])
+                for i in range(n):
+                    ncdf_b = _norm_cdf((_bpnts - pers_dgm[i, 0]) / sigma)
+                    ncdf_p = _norm_cdf((_ppnts - pers_dgm[i, 1]) / sigma)
+                    curr_img = ncdf_p[None, :] * ncdf_b[:, None]
+                    pers_img += wts[i]*(curr_img[1:, 1:] - curr_img[:-1, 1:] - curr_img[1:, :-1] + curr_img[:-1, :-1])
+            else:
+                general_flag = True
+
+        # handle the general case
+        if general_flag:
+            bb, pp = np.meshgrid(_bpnts, _ppnts, indexing='ij')
+            bb = bb.flatten(order='C')
+            pp = pp.flatten(order='C')
+            for i in range(n):
+                curr_img = np.reshape(kernel(bb, pp, mu=pers_dgm[i, :], **kernel_params),
+                                      (resolution[0]+1, resolution[1]+1), order='C')
+                pers_img += wts[i]*(curr_img[1:, 1:] - curr_img[:-1, 1:] - curr_img[1:, :-1] + curr_img[:-1, :-1])
+
+        return pers_img
+"""
```

Comment:
You can put this docstring in the function body like the rest of the docstrings.

Reply:
ohh I see this is a separation for a number of kernel functions.  Could they be moved to a separate file?

Reply:
I moved these to images_kernels.py.